### PR TITLE
Consider designMode and contentEditable

### PR DESCRIPTION
**Pull request by mightyiam**
For now, I'll use `textarea`s for my tests.

This will be useful, though.

### CODE REVIEW

**Comment by eventualbuddha** [comment]
Interesting. Having a test would be great! Also, putting the `return` statement on its own line would be good too.

**Comment by mightyiam** [comment]
Would be much nicer to work on this after #21 is fixed.

File: lib/keysim.js [hunk 1]
```diff
@@ -401,6 +401,9 @@ export class Keyboard {
         return true;
 
       default:
+        if (target.designMode.toLowerCase === 'on' || target.isContentEditable) {
```
Comment (mightyiam) [inline lib/keysim.js]: This is not how `designMode` should be checked. `designMode` should be checked on the `document` property of the provided element.

**Comment by mightyiam** [comment]
The tests should test that:

The methods
 
  1. `#dispatchEventsForInput`
  1. `#dispatchEventsForAction`
  1. `#dispatchEventsForKeystroke`

1. Do not fire events when the element is not editable. Perhaps there is a module for that? Anyway, there should be.
1. Do fire events when the element is editable.

**Comment by eventualbuddha** [comment]
I'm not sure what to do about this one. Are you going to update the PR?

**Comment by mightyiam** [comment]
Sure am.

**Comment by mightyiam** [comment]
@eventualbuddha this should serve most use cases. The new dependency is a new module I wrote just for this. It seems to me to be sufficiently tested so I don't think we require any additional tests here about that functionality. What do you think?

**Comment by mightyiam** [comment]
@eventualbuddha ready for review.

**Comment by mightyiam** [comment]
@eventualbuddha I need you to press the big green button please, Brian.

**Comment by mightyiam** [comment]
Do **not** press the big green button, please! Please merge #29 before this and I'll have some more work to do on this one. Just a little.

**Comment by AnilRedshift** [comment]
Hi @eventualbuddha can we get this merged in? I need to support contenteditable for my project as well.

The tool is super useful otherwise. Thanks!

**Comment by AnilRedshift** [comment]
For anyone who finds it helpful in the meantime, here's the quick hack I put in our tests to make this work:
```
function fixUpKeyboard(keyboard) {
    const proto = Object.getPrototypeOf(keyboard);
    if (keyboard.targetCanReceiveTextInput === proto.targetCanReceiveTextInput) {
        const oldHandler = keyboard.targetCanReceiveTextInput;

        keyboard.targetCanReceiveTextInput = (target) => {
            let allowed = false;
            if (oldHandler.call(keyboard, target)) {
                allowed = true;
            }
            else if (target && target.getAttribute(`contenteditable`) === `true`) {
                allowed = true;
            }
            return allowed;
        };
    }
    return keyboard;
}
```

**Comment by eventualbuddha** [comment]
This should be fixed by the next release that includes #37 (should be v1.4.0).

**Comment by AnilRedshift** [comment]
I can confirm that 2.0 fixes this.

**Pull request closed without merging**